### PR TITLE
File encryption and decryption example ; dev container with c library build and tpm emulator

### PR DESCRIPTION
**Pull request by nicolaspernoud**
Hello,

Here is a pull request with those two features that seems useful...

Best regards.

### CODE REVIEW

**Review by ionut-arm — COMMENTED** [review]
Hey, thanks for the example and the devcontainer! I'd been looking at devcontainers for a while hoping we could have something like this, glad someone put it together :)

One thing to note is that you'll need to add your text file as [an exception](https://github.com/codespell-project/actions-codespell?tab=readme-ov-file#parameter-exclude_file) to the spellchecking.

You probably also need to wait for #532 to be merged :]

Other than that, LGTM

File: tss-esapi/examples/symmetric_file_encrypt_decrypt.rs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        .with_st_clear(false)
+        .with_sensitive_data_origin(true)
+        .with_user_with_auth(true)
+        .with_sign_encrypt(true)
+        .with_decrypt(true)
+        .build()
+        .expect("Failed to build object attributes");
+
+    let key_pub = PublicBuilder::new()
+        // This key is a AES key
```
Comment (Superhepper) [inline tss-esapi/examples/symmetric_file_encrypt_decrypt.rs:45]: Minor:
Should it be 'an AES key'?

**Review by Superhepper — DISMISSED** [review]

File: tss-esapi/examples/symmetric_file_encrypt_decrypt.rs (hunk 1 above)
Comment (nicolaspernoud) [reply tss-esapi/examples/symmetric_file_encrypt_decrypt.rs]: It should... i corrected the typo.

**Comment by nicolaspernoud** [comment]
@ionut-arm and @Superhepper : i made the changes, thanks for your feedbacks !

**Comment by nicolaspernoud** [comment]
I updated using github web "update" button : it merges instead of rebasing, but that should be ok.

**Comment by wiktor-k** [comment]
I think the only missing bit is running `cargo fmt` locally and committing and pushing the result :thinking:

**Comment by ionut-arm** [comment]
Unfortunately I don't think we can merge as is now - you'll need to rebase and get all your commits on top of @Superhepper's, otherwise we'll have his commit twice.

**Review by Superhepper — APPROVED** [review]

**Pull request merged**